blocks/gu_spdetails/block_gu_spdetails.php

```diff
@@ -189,18 +189,12 @@ public static function return_assessments($user) {
     /**
      * Checks if the user has a 'student' role in the course
      *
-     * @param int $userid
      * @param int $courseid
      * @return boolean
      */
-    public static function return_isstudent($userid, $courseid) {
-        $roles_array = array();
-        // get_user_roles_in_course() from /lib/accesslib.php
-        $roles = strtolower(strip_tags(get_user_roles_in_course($userid, $courseid)));
-        $roles_array = explode(',', $roles);
-        $isstudent = in_array(get_string('student', 'block_gu_spdetails'), $roles_array);
-
-        return $isstudent;
+    public static function return_isstudent($courseid) {
+        $context = context_course::instance($courseid);
+        return !has_capability('moodle/course:update', $context);
```

Comment:
We cannot use this capability `moodle/course:update` since this is only limited for the default roles of managers and teachers. See [link](https://docs.moodle.org/310/en/Capabilities/moodle/course:updatehttps://docs.moodle.org/310/en/Capabilities/moodle/course:update) 

We can target the capability by activity instance (mod->instance) and activity type (mod->modname) instead:
Here are the capabilities for each activity type:
Assignment: `/mod/assign:submit`
Forum: `/mod/forum:viewdiscussion`
Quiz: `/mod/quiz:attempt`
Workshop: `/mod/workshop:submit`

To see list of other capabilities, pls check [here](https://docs.moodle.org/310/en/index.php?title=Category:Capabilities).

Reply:
Requested changes are applied to commit [1c64b58](https://github.com/uofg-andf/andfmoodle/pull/304/commits/1c64b5867120b39e399ab469d5af4a95f6e1734a)